x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -302,12 +302,17 @@ public InferenceProcessor create(Map<String, Processor.Factory> processorFactori
                 fieldMap = Collections.emptyMap();
             }
 
-            InferenceConfigUpdate inferenceConfig;
+            InferenceConfigUpdate inferenceConfigUpdate;
             Map<String, Object> inferenceConfigMap = ConfigurationUtils.readOptionalMap(TYPE, tag, config, INFERENCE_CONFIG);
             if (inferenceConfigMap == null) {
-                inferenceConfig = new EmptyConfigUpdate();
+                if (minNodeVersion.before(EmptyConfigUpdate.minimumSupportedVersion())) {
```

Comment:
With this change, I am wondering if `toConfig()` should throw for `EmptyConfigUpdate`. IIRC, `toConfig` is only used for BWC serialization.

Reply:
👍